src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -633,71 +636,79 @@ export class OnlineGameWrapperComponent extends GameWrapper implements OnInit, O
                 return this.notifyVictory(encodedMove, scores);
             }
         } else {
-            const user: Player = this.getPlayer();
+            const player: Player = this.observerRole as Player;
             return this.gameService.updateDBBoard(this.currentPartId,
-                                                  user,
+                                                  player,
                                                   encodedMove,
                                                   msToSubstract,
                                                   scores);
         }
     }
     public async resign(): Promise<void> {
+        display(OnlineGameWrapperComponent.VERBOSE, 'OnlineGameWrapperComponent.resign');
         const lastIndex: number = this.getLastIndex();
-        const user: Player = this.getPlayer();
-        const resigner: MinimalUser = this.players[this.observerRole % 2].get();
-        const victoriousOpponent: MinimalUser = this.players[(this.observerRole + 1) % 2].get();
-        await this.gameService.resign(this.currentPartId, lastIndex, user, victoriousOpponent, resigner);
+        const player: Player = this.observerRole as Player;
+        const resigner: MinimalUser = this.players[this.observerRole.value % 2].get();
```

Comment:
du coup là, c'est this.getPlayer(), le resigner !

Reply:
changé !